packages/cli/README.md

```diff
@@ -1,66 +1,478 @@
+
+Files for entry-point commands typically aren't too complicated. Here's the contents of `generate.js` in its entirety:
+
+```javascript
+export const command = 'generate <type>'
+export const aliases = ['g']
+export const description = 'Save time by generating boilerplate code'
+import terminalLink from 'terminal-link'
+
+export const builder = (yargs) =>
+  yargs
+    .commandDir('./generate', { recurse: true })
+    .demandCommand()
+    .epilogue(
+      `Also see the ${terminalLink(
+        'Redwood CLI Reference',
+        'https://redwoodjs.com/reference/command-line-interface#generate-alias-g'
+      )}`
+    )
+
+```
+
+The call to `commandDir` is what makes this command an entry point. You can read more about `commandDir` [here](https://github.com/yargs/yargs/blob/master/docs/advanced.md#commanddirdirectory-opts).
+
+> The second argument to `commandDir`, `{ recurse: true }` is specific to the `generate` command because each of the commands in the `generate` directory is in its own directory to keep things organized--more on this in [Adding a Generator](#adding-a-generator).
+
+Now all the commands in the `generate` directory will be arguments to the `generate` command:
+
+```
+./src/commands/generate
+├── auth
+├── cell
+├── component
+├── function
+├── helpers.js
+├── layout
+├── page
+├── README.md
+├── scaffold
+├── sdl
+├── service
+└── __tests__
+```
+
+There's files and directories here that aren't yargs related&mdash;`README.md`, `helper.js`, and `__tests__`, but because yargs will only use the files that export the appropriate constants, it's ok.
+
+Compare the `generate` command to [redwood-tools](https://github.com/redwoodjs/redwood/blob/master/packages/cli/src/redwood-tools.js)&mdash;another entry point command, but one that has all its logic in a single file.
+
+### Adding a Generator
+
+> We're about to refactor generators out of @redwoodjs/cli and into their own package, so a lot of this section will probably change soon.
+
+You can add a generator by creating a directory and file in `./src/commands/generate`
+Although it's not necessary, the directory and file should be named after the command that invokes it (for consistency).
+For example, the page generator, which is invoked with
+
+```
+yarn redwood generate page 
+```
+
+lives in `./src/commands/generate/page/page.js`, where the `page` directory has the following structure:
 
 ```terminal
-$ yarn create redwood-app ~/myprojects/todo
+src/commands/generate/page
+├── page.js
+├── templates
+└── __tests__
 ```
 
-## Development
+Since a typical generator writes files, needs templates to do so, and needs tests to ensure it works, we use this command-in-a-directory structure to keep things organized. The entry-point command, `generate.js`, will stil be able to find these command files (here `page.js`) because of the second argument (`{ recurse: true }`) to `commandDir`.
```

Comment:
```suggestion
Since a typical generator writes files, needs templates to do so, and needs tests to ensure it works, we use this command-in-a-directory structure to keep things organized. The entry-point command, `generate.js`, will still be able to find these command files (here `page.js`) because of the second argument (`{ recurse: true }`) to `commandDir`.
```